src/libutil/file-system.cc

```diff
@@ -21,8 +21,14 @@ namespace fs = std::filesystem;
 
 namespace nix {
 
-Path absPath(Path path, std::optional<PathView> dir, bool resolveSymlinks)
+Path absPath(PathView path, std::optional<PathView> dir, bool resolveSymlinks)
 {
+    // If we set `path` to refer to a newly-created string, we set
+    // `scratch` to that string first, and then set `path` to `scratch`. This ensures
+    // the newly-created string lives long enough for the call to
+    // `canonPath`, and allows us to just accept a `std::string_view`.
+    std::string scratch;
```

Comment:
Maybe move this into the conditional?
If the path is already absolute, we don't need to initialize it `scratch`.

Reply:
IIRC an empty string is very cheap to initialize so it doesn't really matter.

Reply:
Oh I thought you meant move the comment. The string has to be out side so it lives long enough.

The alternative is to duplicate the call and then we don't need `scratch`.